### ISSUE TITLE
Scrolling using keyboard and past the line

### ISSUE THREAD

**Issue by yashpalgoyal1304** [wez/wezterm#2224]
### Is your feature request related to a problem? Please describe
* I cant scroll using keyboard shortcuts --> have to use the mouse/touchpad
* Also, the command prompt gets stuck in the bottom making it uneasy on the eye

### Describe the solution you'd like
Ability to:
1. scroll through the screen via some keyboard shortcut like <kbd>Ctrl</kbd> + <kbd>arrow keys</kbd>, and 
2. scroll past the last line upto the point that only last line is still visible on the screen

References:
both of above are supported in:
* text editors (like notepad3 `^arrow`, geany `alt-arrow`, & vscodium `^arrow`)
* windows conhost `^arrow` too, though that is a bit extreme and supports scrolling endlessly - which aint wanted either

Note: in editors too, scrolling the screen/viewerport is different than changing the cursor position

### Describe alternatives you've considered
`clear` and `^l` - which clears the screen, but sometimes i dont want to alter the contents, just scroll

(`^` = <kbd>Ctrl</kbd>)

### Versions:
* OS: windows 10 20H2
* wezterm: 20220624-141144-bd1b7c5d

**Comment by yashpalgoyal1304** [comment]
kinda relevant maybe: https://github.com/wez/wezterm/issues/427

**Comment by wez** [comment]
There are a number of key assignments for scrolling:
[ScrollByLine](https://wezfurlong.org/wezterm/config/lua/keyassignment/ScrollByLine.html)
[ScrollByPage](https://wezfurlong.org/wezterm/config/lua/keyassignment/ScrollByPage.html) (SHIFT+PageUp/PageDown are bound to this by default)
[ScrollToBottom](https://wezfurlong.org/wezterm/config/lua/keyassignment/ScrollToBottom.html)
[ScrollToPrompt](https://wezfurlong.org/wezterm/config/lua/keyassignment/ScrollToPrompt.html)
[ScrollToTop](https://wezfurlong.org/wezterm/config/lua/keyassignment/ScrollToTop.html)

**Comment by yashpalgoyal1304** [comment]
is any of these setup by default i.e. available to try directly or will i have to set those up?

**Comment by wez** [comment]
Please read my comment again; it explains about the defaults.
The full list of defaults can be found at https://wezfurlong.org/wezterm/config/default-keys.html

It is not possible to scroll past the last line, as it is the last line and there are no lines after it.

**Comment by bew** [comment]
FWIW I also like to 'scroll'  past the last line a bit (I use it to 'clear' my mind, give the screen some space), the way I do it is that in my shell I have mapped a key to send a few ANSI escape sequences to scroll the content of the terminal up.
See: https://github.com/bew/dotfiles/blob/1e1117fec1a043027a40e27434acf85d5fdeb00d/zsh/rc/mappings.zsh#L86

Note: that I'd love to be able to scroll it back down while keeping the scrollback, however no escape sequence can do that.
Kitty implemented one, and I asked to have it in westerm some time ago, not impl so far (maybe I'll just do ot some day)..
See:  #1513

**Comment by MuhammedZakir** [comment]
@bew: Thanks for that snippet! This is what I came up with: `seq 3 `tput lines`; tput clear;`. Set `3` to `2` when not invoking manually, using a key binding for example.

**Comment by yashpalgoyal1304** [comment]
here's the recording video with the editors and windows conhost terminal, 
_speed it up while watching to save time, sorry with excessive scrolling and jittery feel of the vid_

https://user-images.githubusercontent.com/19423063/178258999-d71e56b0-53f9-43c6-be8b-94a52401b0c8.mp4

**Comment by yashpalgoyal1304** [comment]
oh i forgot to show this, conhost has this property toggleable too, by the name of `Disable scroll forward`

![image](https://user-images.githubusercontent.com/19423063/178640246-1437d5af-f053-401f-98cb-972fdf867484.png)

**Comment by MuhammedZakir** [comment]
There is an issue with solution posted by @*bew*, and myself, it doesn't work when a program is running. This won't be a problem if there is a terminal-native option.

**Comment by goyalyashpal** [comment]
lack of this in the terminals also make the video tutorials super hard to follow with subtitles turned on

also bad if you are a content creator for such tutorials...

as can be seen in this screenshot took from this video: [CpTfQ-q6MPU](https://www.youtube.com/watch?v=CpTfQ-q6MPU "'10 Linux Terminal Commands for Beginners' by Gary Explains")

![image](https://github.com/wez/wezterm/assets/19423063/31682f4a-228d-49f0-8af5-e9a847122ba5)
